lib/device_detector/parser.rb

```diff
@@ -37,32 +37,43 @@ def filenames
 
     def filepaths
       filenames.map do |filename|
-        File.join(ROOT, 'regexes', filename)
+        [ filename.to_sym, File.join(ROOT, 'regexes', filename) ]
       end
     end
 
-    def regexes_for(filepaths)
+    def regexes_for(file_paths)
       from_cache(['regexes', self.class]) do
-        raw_regexes = load_regexes
-        parsed_regexes = raw_regexes.map { |r| parse_regexes(r) }
-
-        parsed_regexes.flatten
+        load_regexes(file_paths).flat_map { |path, regex| parse_regexes(path, regex) }
       end
     end
 
-    def load_regexes
-      raw_files = filepaths.map { |path| File.read(path) }
+    def load_regexes(file_paths)
+      file_paths.map { |path, full_path| [path, symbolize_keys!(YAML.load_file(full_path))] }
+    end
 
-      raw_files.map { |f| YAML.load(f) }
+    def symbolize_keys!(object)
```

Comment:
The loaded yaml structures will use a lot less memory as a symbol is a so called immediate value. Access will also be faster as hashing/comparing a symbol is much faster than comparing strings. And I find the code more readable.